conveyal/analysis-ui
# allow basemap style to be set with NEXT_PUBLIC_MAPBOX_STYLE

## Description

Currently the frontend can't be run by anyone without a Conveyal-issued Mapbox key, because the default Mapbox style is private. This allows overriding the basemap style by setting the environment variable NEXT_PUBLIC_MAPBOX_STYLE to a fully-qualified mapbox style name (e.g. `mapbox/light-v10`).

I used the NEXT_PUBLIC_ prefix rather than edit `next.config.js` as this seems preferred for recent versions of next.

The environment variables don't seem to be documented anywhere, but happy to document this variable if some points me where to do it.

## How to test

1. Set `MAPBOX_ACCESS_TOKEN` to a token that isn't from Conveyal.
2. Load Analysis, note that it errors loading the basemap (basemap is just gray).
3. Set `NEXT_PUBLIC_MAPBOX_STYLE=mapbox/light-v10`, restart, and observe that the basemap now works.
